lib/message/decrypt.js

```diff
@@ -49,30 +49,35 @@ export async function decryptMessage({ date = serverTime(), encryptedSignature,
 
 // Backwards-compatible decrypt message function
 // `options.message` must be a string, to properly handle legacy messages (and avoid misusing this function)
-export async function decryptMessageLegacy({ messageDate, ...options }) {
-    const sanitizedOptions = { ...options };
+export async function decryptMessageLegacy({ messageDate, message, decryptionKeys, ...options }) {
+    if (!isString(message)) {
+        throw new Error('Armored message expected. Use `decryptMessage` for binary messages or Message objects.')
+    }
 
     if (messageDate === undefined || !(messageDate instanceof Date)) {
         throw new Error('Missing message time');
     }
 
-    let oldEncMessage = getEncMessageFromEmailPM(sanitizedOptions.message);
-    const oldEncRandomKey = getEncRandomKeyFromEmailPM(sanitizedOptions.message);
+    let oldEncMessage = getEncMessageFromEmailPM(message);
+    const oldEncRandomKey = getEncRandomKeyFromEmailPM(message);
 
-    // OpenPGP
+    // Standard OpenPGP message
     if (oldEncMessage === '' || oldEncRandomKey === '') {
         // Convert message string to object
-        sanitizedOptions.message = await getMessage(sanitizedOptions.message);
-        return decryptMessage(sanitizedOptions);
+        return decryptMessage({
+            message: await getMessage(message),
```

Comment:
And then here, since we know it's an armored message, we can just do `readMessage({ armoredMessage })`, then we don't have to check the type again. In fact, can we get rid of `getMessage`, `getSignature` and `getCleartextMessage`? (Doesn't have to happen in this PR)